web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -55,6 +55,35 @@ export class DataStoreService {
       );
   }
 
+  /**
+   * Returns an Observable that loads and emits the projects to which the user
+   * is part of.
+   *
+   */
+  loadAccessibleProject$(userEmail: string): Observable<Project[]> {
+    return this.db
+      .collection('projects')
+      .snapshotChanges()
+      .pipe(
+        map(actions =>
+          actions
+            .filter(a => {
+              const docData = a.payload.doc.data() as DocumentData;
+              const emails = Object.keys(docData.acl);
+              return emails.indexOf(userEmail) > -1;
+            })
+            .map(a => {
+              const docData = a.payload.doc.data() as DocumentData;
+              const id = a.payload.doc.id;
+              return FirebaseDataConverter.toProject(
+                id,
+                docData as DocumentData
```

Comment:
You don't have to do `as DocumentData` again since you did it once already above.

Reply:
updated